common/cuda_hip/factorization/factorization_kernels.cpp

```diff
@@ -588,12 +588,23 @@ void symbolic_validate(
             factors_lookup.storage.get_const_data(), found.get_data(),
             missing.get_data());
     }
+    auto get_bool_identity = []() {
```

Comment:
should we perhaps just implement https://en.cppreference.com/w/cpp/utility/functional/identity.html in our `std_extension.hpp` until we can use the c++20 version? Then we don't need to check for the cuda version.

Reply:
I may still go for this now when we only use it here.
I have seen their code has `__is_identity` and used in somewhere. It might be useful when they can skip this to avoid another layer code and help the compiler optimization.
std::identity will be not in this case